src/EnsembleKalmanInversions.jl

```diff
@@ -122,36 +131,57 @@ function EnsembleKalmanInversion(inverse_problem;
                                  unconstrained_parameters = nothing,
                                  forward_map_output = nothing,
                                  mark_failed_particles = NormExceedsMedian(1e9),
-                                 ensemble_kalman_process = Inversion())
+                                 ensemble_kalman_process = Inversion(),
+                                 tikhonov = false)
 
     if ensemble_kalman_process isa Sampler && !isnothing(pseudo_stepping)
         @warn "Process is $ensemble_kalman_process; ignoring keyword argument pseudo_stepping=$pseudo_stepping."
         pseudo_stepping = nothing
     end
 
+    free_parameters = inverse_problem.free_parameters
+    priors = free_parameters.priors
+    Nθ = length(priors)
+    Nens = Nensemble(inverse_problem)
+
+    # Generate an initial sample of parameters
+    unconstrained_priors = NamedTuple(name => unconstrained_prior(priors[name])
+                                      for name in free_parameters.names)
+
     if isnothing(unconstrained_parameters)
         isnothing(forward_map_output) ||
             throw(ArgumentError("Cannot provide forward_map_output without unconstrained_parameters."))
 
-        free_parameters = inverse_problem.free_parameters
-        priors = free_parameters.priors
-        Nθ = length(priors)
-        Nens = Nensemble(inverse_problem)
-
-        # Generate an initial sample of parameters
-        unconstrained_priors = NamedTuple(name => unconstrained_prior(priors[name])
-                                          for name in free_parameters.names)
-
         unconstrained_parameters = [rand(unconstrained_priors[i]) for i=1:Nθ, k=1:Nens]
     end
 
     # Build EKP-friendly observations "y" and the covariance matrix of observational uncertainty "Γy"
     y = dropdims(observation_map(inverse_problem), dims=2) # length(forward_map_output) column vector
-    Γy = construct_noise_covariance(noise_covariance, y) # noise_covariance * UniformScaling(1.0)
+    Γy = construct_noise_covariance(noise_covariance, y)
     Xᵢ = unconstrained_parameters
     iteration = 0
     pseudotime = 0.0
 
+    # Pre-compute Γθ^(-1/2) and μθ
+    unconstrained_priors = collect(unconstrained_priors)
+    Γθ = diagm(getproperty.(unconstrained_priors, :σ).^2)
+    μθ = getproperty.(unconstrained_priors, :μ)
+
+    precomputed_arrays = Dict(:inv_Γy => inv(Γy), 
+                              :inv_sqrt_Γy => inv(sqrt(Γy)),
+                              :Γθ => Γθ,
+                              :inv_sqrt_Γθ => inv(sqrt(Γθ)),
+                              :μθ => μθ)
+
+    Σ = cat(Γy, Γθ, dims=(1,2))
+    precomputed_augmented_arrays = Dict(:y_augmented => vcat(y, zeros(Nθ)),
+                                        :η_mean_augmented => vcat(zeros(length(y)), -μθ),
+                                        :Σ => Σ, 
+                                        :inv_Σ => inv(Σ),
+                                        :inv_sqrt_Σ => inv(sqrt(Σ)))
```

Comment:
Can you add a comment that documents what each of these properties mean? When we have a type that represents this data, we can add that to the docstring.

Reply:
If I were to implement an object with non-symbol attributes to store all the information in `precomputed_arrays`, it would look something like this:

```julia
struct PrecomputedArrays{M, T}
        inverse_observation_noise_covariance :: M
        inverse_sqrt_observation_noise_covariance :: M
        prior_covariance :: M
        inverse_sqrt_prior_covariance :: M
        prior_means :: T
        augmented_mapped_observation :: T
        augmented_observation_noise_mean :: M
        augmented_observation_noise_covariance :: M
        inverse_augmented_observation_noise_covariance :: M
        inverse_sqrt_augmented_observation_noise_covariance :: M
end
```

Reply:
Can you also report which pseudo-time-stepping methods require which data?